project/api/models.py

```diff
@@ -87,9 +90,15 @@ def clean(self, *args, **kwargs):
         if self.start_date > self.end_date:
             raise ValidationError("Start date cannot be later than end date")
         if loads(self.geo.json)["type"] != "Polygon" and loads(self.geo.json)["type"] != "MultiPolygon":
-            raise ValidationError("Only Polygon and MultiPolygon objects are acceptable geometry types")
+            raise ValidationError(
+                "Only Polygon and MultiPolygon objects are acceptable geometry types")
         super(Territory, self).clean(*args, **kwargs)
 
+        # This date check is inculsive.
+        if Territory.objects.filter(start_date__range=(self.start_date, self.end_date), nation=self.nation).exists() or Territory.objects.filter(end_date__range=(self.start_date, self.end_date), nation=self.nation).exists():
```

Comment:
Still incorrect. If a territory with [1;5] already exists, and I create a new one with [2;3], it should raise an error. It does not wit this logic.

Reply:
With the 2nd filter being there, we can remove this one.